cirq/ops/raw_types.py

```diff
@@ -206,6 +238,13 @@ def wrap_in_linear_combination(
         from cirq.ops import linear_combinations
         return linear_combinations.LinearCombinationOfGates({self: coefficient})
 
+    def add_tag(self, tag: Tag):
+        self._tags = getattr(self, '_tags', [])
```

Comment:
using `getattr` seems really odd and will likely confuse my dear pycharm. Why not initialize to `[]` in `__init__`?

Reply:
Because Gate is an abstract type and doesn't define __init__.  We could do this, but then every other Gate implementation has to call super.__init__() which it doesn't right now.

Reply:
oooo

Reply:
this might actually be worth putting in as a comment to prevent someone from wasting time in the future